### DISCUSSION TITLE
Release Notes CovPass and CovPassCheck Version 1.17

### DISCUSSION THREAD

**Discussion by alexcimander** [Digitaler-Impfnachweis/covpass-ios#121] [Announcements]
📢 **Release Notes CovPass and CovPassCheck Version 1.17** 📢  
Version 1.17 of the CovPass and CovPassCheck is planned to be released early february. It will include a variety of features and fixes that were requested by the community.

**Planned Main Features CovPass:**
•	Improvements on Trustlist-Updates: The apps will now use etag, gzip and other http-features to improve the update-mechanisms of the trustlist (Already present for IOS / New for Android).

**Planned Main Features CovPassCheck:**
•	The CovPassCheck now offers scan-sequences to check your 2G+ Status. The awaited booster-recognition, especially regarding the use-case “valid test not necessary after booster vaccination” is planned to be added with the next release 1.18.

**Planned Bug-Fixes:**
•	Multiple UI-Bugfixes as Accessibility-Enhancements are planned for the release.  
  
Thank you all for your feedback, ideas and critics!

stay safe and best regards,

Team CovPass and CovPassCheck ✔️

**Comment by Ein-Tim** [comment]
Are the features only "Planned" or are they actually in the release? (-;
If they are in the release, the wording should be changed.

**Comment by timokoenig** [comment]
>  The CovPassCheck now offers scan-sequences to check your 2G+ Status. The awaited booster-recognition, especially regarding the use-case “valid test not necessary after booster vaccination” is planned to be added with the next release 1.18.

Meaning the CheckApp is still not fully functional for the current 2G+ case? 😞 

Does CovPass at least show the wording booster to make it easier to recognise?

**Comment by Kurt57** [comment]
1.17 was released yesterday. Including 2G+. The repository update will follow today.

**Reply by aherberger** [reply]
How das the App define 2G+ - for example here in Hessen 2G+ has a very complicated definition, so relying on the CovPassCheck 2G+ message would lead to blocking people from entrance (or aksing for a test if not needed), that are basically allowed for entrance (for example a double vaccinated person where the 2nd dose is allowed to enter without a test for 3 month after the 2nd vaccination. For details look here: https://www.hessen.de/sites/hessen.hessen.de/files/2022-01/2g-plusregeln_2101_v5.pdf
Is there any documentation about the rules, that covpasscheck applies for deciding if somebody fulfills the 2G-Rules?

**Reply by Ein-Tim** [reply]
Currently CovPassCheck always shows that a test is required there is no detection of boosters implemented yet.

**Reply by aherberger** [reply]
Thanks for the answer - but how does it come, that this mode is than called "2g+" - or otherwise does 2g+ in CovPassCheck-Terms mean "Immunziation required plus Test" - independend of the status of the immunization? 
What rules will be followed in  the planned booster-update?

**Reply by Ein-Tim** [reply]
@aherberger 

This discussion says:

> The awaited booster-recognition, especially regarding the use-case “valid test not necessary after booster vaccination” is planned to be added with the next release 1.18.

**Reply by Dale81** [reply]
@aherberger 

I assume, that the concept does not include the consideration of "local" rules, so it might be only like this:

- 3/3, 2/1, 3/1 etc. --> Recognize as booster --> No Test needed because it's 2G+ equivalent (as booster is like 2G+ in whole Germany)
- All other possible topics, e.g. last vaccination not older than 90 days --> 2G+ are so "Bundesland-specific" that they are not represented.

Also it seems that even Germany wide rules are not implemented properly as Johnson & Johnson with one dose are still represented as "vaild" for 2G (but they should not be valid as of current rules for "vollständig geimpft/fully vaccinated"). Thats a big problem as you can't rely on the app. In no case because it could ALWAYS be, that the person you are scanning has a not-valid 2G vaccination.

Out of my perspective, the app needs a functionality that whenever new rules are coming out (even shortly on a weekend) the rules get updated and represented in the app. Would be great to get some feedback on this from the developers. @alexcimander

**Reply by alexcimander** [reply]
When it comes to rules they are independent of the releases. So whenever the ruleset is older than 24h the apps will check for updates. If new rules are present (defined and hosted by RKI and TSI) those rules can be downloaded. However since the rules are currently in constant change their digital representations have a hard time keeping up with the pace. 

Besides that the Rule Engine that checks those rules and even the rules themselves are based on the EU Specifications that are stating rules can only be applied to "one certificate provided". It is not possible to make references to other certifcates etc. 

Saying that rules cannot be used to check on "admission states" like 2G+ since those concepts need some references or context for the presented certificates. This is where the fun begins. 

Since we have 16 federal states that are changing there admission states constantly and rules cannot be part of the solution for those admission states we need to design the check-app in a way that all information required will be displayed to the users that they can check the certificates. Doing that we need to be careful that we present the info in a way that we cannot simply grant access based on one certificate (if this certificate needs an companion in another federal state). 

The situation is super tricky for us but we are trying our best to keep up with all the rule changes and admission state requirements.

**Reply by alexcimander** [reply]
Release 1.18 is mainly focused on the booster as an alternative to a test certificate since this "case" is now valid within all federal states. Release 1.19 will then provide almost all scan sequences which are needed to cover all "special cases" within germany in an universal scan flow. The different meaning of 2/1 is another issue we are facing and working on.

The problem with 2/1 certificates now is that they have different "value" and no historic information in place. A 2/1 after JJ base immunization is now considered as "base immunization" as for 2/1 certificates issued after vaccination after recovery is considered as "booster". 

Rules can only check "2 out of 1? If yes: valid" but they simply cannot take care of the historic context. Besides that now information about this context is presented within the certificate itself. 

Only solution for now is to manually check "historic" certificates. Saying that we are working and investigating on solutions how these certificates can be presented in a feasible and understandable way to the CovPassCheck to grant access to different admission states. We will keep you updated about it.

**Comment by aherberger** [comment]
Falls jemand Interesse hat und gerne eher auf Deutsch kommunizieren möchte und in Kontakt zu Anwender kommen - ich habe mal eine Facebook-Gruppe zum Erfahrungsaustausch rund um CovPassCheck gestartet, denn die Anwender selbst werden eher weniger hier auf Github vorbeikommen... https://www.facebook.com/groups/1864479393747104
